src/network/instance.ts

```diff
@@ -0,0 +1,74 @@
+import axios, { AxiosError, AxiosResponse } from 'axios';
+
+import { ErrorHandler } from './errorHandler';
+
+import ApiConstants from './apiConstants';
+
+const { BASE_URL } = ApiConstants;
+
+const instance = axios.create({
+  baseURL: BASE_URL,
+});
+
+function interceptorResponse<T>(
+  response: AxiosResponse<ApiCallResponse<T>>,
+): AxiosResponse<ApiCallResponse<T>> {
+  if (!response.data) {
+    throw new Error(response.statusText || 'Unknown error occurred');
+  }
+
+  const apiSuccess: ApiCallSuccess<T> = { success: true, responseData: response.data as T };
+  return { ...response, data: apiSuccess };
+}
+
+function interceptorError(error: AxiosError): ApiCallFailure {
+  const apiError = ErrorHandler(error);
+
+  const apiFailure: ApiCallFailure = { success: false, error: apiError };
+  return apiFailure;
+}
+
+instance.interceptors.response.use(interceptorResponse, interceptorError);
+
+async function _get<T, Params = {}>(
```

Comment:
Generally, it's preferred to put the CRUD call functions in a separate file, since the file is only for the Axios instance keep the instance setup here and move other things to a different file.